internal/clusterfeature/clusterfeatureadapter/gorm_feature_repository.go

```diff
@@ -144,3 +155,45 @@ func (r *GormFeatureRepository) modelToFeature(cfm *clusterFeatureModel) (*clust
 
 	return &f, nil
 }
+
+// DeleteFeature permanently deletes the feature record
+func (r *GormFeatureRepository) DeleteFeature(ctx context.Context, clusterId uint, featureName string) error {
+
+	fm := clusterFeatureModel{
+		ClusterID: clusterId,
+		Name:      featureName,
+	}
+
+	if err := r.db.Delete(&fm).Error; err != nil {
+		return emperror.Wrap(err, "could not delete status")
+	}
+
+	return nil
+
+}
+
+func (r *GormFeatureRepository) ListFeatures(ctx context.Context, clusterId uint) ([]*clusterfeature.Feature, error) {
+	var (
+		featureModels []clusterFeatureModel
+	)
+
+	featureList := make([]*clusterfeature.Feature, 0)
```

Comment:
Why create an empty slice? `nil` is easier:
```go
var featureList []*clusterfeature.Feature
```
This way, you could merge the declarations and using the parentheses would make sense.

Reply:
fixed, thx